scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,371 @@
+      IO.println
+        s!"mkRenamesDict: '{pct}' should have been of the form Rxxx, denoting a `R`ename \
+          and a similarity percentage.\nFull git line: '{git}'"
+      continue
+    let pctNat := (pct.drop 1).toNat?.getD 0
+    -- This looks like a rename with a similarity index at least as big as our threshold:
+    -- we add the rename to our dictionary.
+    if percent ≤ pctNat then
+      dict := dict.insert oldName newName
+    -- This looks like a rename, but the similarity index is smaller than our threshold:
+    -- we report a message and do not add the rename to our dictionary.
+    else
+      IO.println
+        s!"'{oldName}' was renamed to '{newName}' ({pct}), but the similarity {pctNat}% \
+          is less than the expected threshold of {percent}%.\n\n
+          We treat this file as a removal."
+  return dict
+
+/--
+`mkModName fname` takes as input a file path and returns the guessed module name:
+the dir-separators get converted to `.` and a trailing `.lean` gets removed, if it exists.
+-/
+def mkModName (fname : System.FilePath) : String :=
+  let cpts := fname.components
+  let cpts :=
+    match cpts.getLast? with
+    | none => cpts
+    | some last =>
+      cpts.dropLast ++ [if last.endsWith ".lean" then last.dropRight ".lean".length else last]
+  ".".intercalate cpts
+
+#guard mkModName ("Mathlib" / "Data" / "Nat" / "Basic.lean") == "Mathlib.Data.Nat.Basic"
+
+/--
+`deprecateFilePath fname comment` takes as input
+* the path `fname` of a file that was deleted;
+* an optional comment to add in the `deprecated module` syntax.
+
+It returns a pair consisting of
+* an array of `MessageData` giving details about the last time that `fname` was modified
+  and when it was deleted;
+* the content of the deprecated file, matching the original `fname` up to the last import command
+  followed by the `deprecated_module` command with the optional `comment` input, defaulting to
+  `Auto-generated deprecation` if `comment = none`.
+-/
+def deprecateFilePath (fname : String) (rename comment : Option String) :
+    CommandElabM (Array MessageData × String) := do
+  let mut msgs : Array MessageData := #[]
+  -- Check that the input `fname` is a file that currently does not exist.
+  if ← System.FilePath.pathExists fname then
+    throwError m!"The file {fname} exists: I cannot deprecate it!"
+  -- Retrieve the last two commits that modified `fname`:
+  -- the last one is the deletion, the previous one is the last file modification.
+  let log ← runCmd s!"git log --pretty=oneline -2 -- {fname}"
+  let [deleted, lastModified] := log.trim.splitOn "\n" |
+    throwError "Found {(log.trim.splitOn "\n").length} commits, but expected 2! \
+      Please make sure the file {fname} actually exists"
+  let (_deleteHash, deletedMsg) ← processPrettyOneLine deleted "deleted" fname
+  let (modifiedHash, modifiedMsg) ← processPrettyOneLine lastModified "last modified" fname
+  msgs := msgs.push <| m!"The file {fname} was\n"
```

Comment:
How about using msgs.append with a three-item array? (Under the hood, that is equivalent --- but perhaps more concise?)